### PR TITLE
Feature/button favorite ddfsoeg 141 V1

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### This is the first version

It only checks if id is on used list, the next version will also be able to add / remove using orval auto generated code

This show that we can using orval auto generate code to help retrieve data to see if a user has a materials on his list. and introduce the button favorite

#### Link to issue
https://reload.atlassian.net/browse/DDFSOEG-142

#### Description
The button favorite accepts an id and if that id is on the used favorite list the fill prop will change. 

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: orval.config.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (cableman) [inline orval.config.ts:1]: This config file should be moved into `src/core/material-list-api` as every generated client needs to have there own config file. Then run orval with `--config` alternative rename the file to `material-list-api.orval.config.ts`?

File: package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "test": "cypress run",
     "cypress:open": "cypress open",
     "codegen:dbc:gateway": "CODEGEN_SCHEMA_AGENCY=100200 CODEGEN_SCHEMA_PROFILE=opac graphql-codegen --config dbc-gateway.codegen.yml",
+    "codegen:rest-services": "orval",
```
Comment (cableman) [inline package.json:38]: Add `--config <FILE>` to use the one that matches this client.

File: src/apps/hello-world/hello-world.entry.tsx [hunk 3]
```diff
@@ -8,8 +8,6 @@ export interface HelloWorldEntryProps {
   whatText: string;
 }
 
-const HelloWorldEntry: React.FC<HelloWorldEntryProps> = (
-  props: HelloWorldEntryProps
-) => <HelloWorld />;
+const HelloWorldEntry: React.FC<HelloWorldEntryProps> = () => <HelloWorld />;
```
Comment (cableman) [inline src/apps/hello-world/hello-world.entry.tsx:11]: Why do `HelloWord` change by adding an API client?

**Review by cableman — CHANGES_REQUESTED** [review]

File: orval.config.ts (hunk 1 above)
Comment (cableman) [reply orval.config.ts:1]: Forget about this one @kasperg has just made me adware that it is possible to have more than one client defined in the same config file :-)

File: package.json (hunk 2 above)
Comment (kasperg) [reply package.json:38]: I will resolve this based on the comment above.

File: src/apps/hello-world/hello-world.entry.tsx (hunk 3 above)
Comment (kasperg) [reply src/apps/hello-world/hello-world.entry.tsx:11]: I think it is leftovers from #30 where the generated client was used in the Hello World app.

**Pull request merged**